src/Containers/CadView.jsx

```diff
@@ -161,62 +190,52 @@ export default function CadView({
   }
 
 
-  /**
-   * Analyze loaded IFC model to configure UI elements.
-   * @param {Object} rootElt Root of the IFC model.
-   * @param {Object} viewer
-   * @param {string} pathLoaded
-   */
-  function onModelLoad(rootElt, viewer, pathLoaded) {
-    debug().log('CadView#onModelLoad...', rootElt)
-    gtag('event', 'select_content', {
-      content_type: 'ifc_model',
-      item_id: pathLoaded,
-    })
-    setRootElement(rootElt)
+  /** Analyze loaded IFC model to configure UI elements. */
+  async function onModel() {
+    if (model == null) {
+      return
+    }
+    const rootElt = await model.ifcManager.getSpatialStructure(0, true)
+    if (rootElt.expressID == undefined) {
+      throw new Error('Model has undefined root express ID')
+    }
     setupLookupAndParentLinks(rootElt, elementsById)
     setDoubleClickListener()
-    initSearch(rootElt, viewer)
+    initSearch(model, rootElt)
+    setRootElement(rootElt)
     setShowNavPanel(true)
   }
 
 
   /**
-   * @param {Object} rootElt Root ifc elment for recursive indexing.
-   * @param {Object} viewer The IfcViewerAPI instance.
+   * Index the model starting at the given rootElt, clearing any
+   * previous index data and parses any incoming search params in the
+   * URL.  Enables search bar when done.
+   * @param {Object} m The IfcViewerAPI instance.
+   * @param {Object} rootElt Root ifc element for recursive indexing.
    */
-  function initSearch(rootElt, viewer) {
+  function initSearch(m, rootElt) {
     searchIndex.clearIndex()
-    debug().log('CadView#initSearch')
+    debug().log('CadView#initSearch: ', m, rootElt)
     debug().time('build searchIndex')
-    searchIndex.indexElement(rootElt, viewer)
+    searchIndex.indexElement(m, rootElt)
     debug().timeEnd('build searchIndex')
-    debug().log('searchIndex: ', searchIndex)
-    onSearch()
+    onSearchParams()
     setShowSearchBar(true)
   }
 
 
-  /** Clear active search state and unpick active scene elts. */
-  function clearSearch() {
-    setSelectedElements([])
-    if (viewer) {
-      viewer.IFC.unpickIfcItems()
-    }
-  }
-
-
   /**
    * Search for the query in the index and select matching items in UI elts.
    * @param {string} query The search query.
```

Comment:
Query is obtained through parsing of the URL, not passed into onSearch function.Right?